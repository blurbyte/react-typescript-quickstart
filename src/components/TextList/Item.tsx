import styled from '@styles/styledComponents';

import Text from '../Text';

const ListItem = Text.withComponent('li');

const Item = styled(ListItem)`
  margin: 0.2rem 0;
`;

export default Item;
