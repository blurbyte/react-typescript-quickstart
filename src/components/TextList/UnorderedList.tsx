import styled from '@styles/styledComponents';
import { Color } from '@styles/colors';

export interface UnorderedListProps {
  simple?: boolean;
}

const UnorderedList = styled<UnorderedListProps, 'ul'>('ul')`
  margin: 0;
  padding: ${props => (props.simple ? 0 : 'auto')};
  color: ${Color.DarkGray};
  list-style-type: ${props => (props.simple ? 'none' : 'square')};
`;

export default UnorderedList;
