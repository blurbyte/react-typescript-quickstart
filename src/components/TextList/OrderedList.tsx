import styled from 'styled-components';

import UnorderedList from './UnorderedList';

const List = UnorderedList.withComponent('ol');

const OrderedList = styled(List)`
  list-style-type: decimal;
`;

export default OrderedList;
