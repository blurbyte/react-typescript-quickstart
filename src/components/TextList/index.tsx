import * as React from 'react';

import UnorderedList from './UnorderedList';
import OrderedList from './OrderedList';
import Item from './Item';

export interface TextListProps {
  ordered?: boolean;
  simple?: boolean;
  children?: React.ReactNode;
}

class TextList extends React.Component<TextListProps, {}> {
  static Item = Item;

  render() {
    const { children, ordered, simple } = this.props;
    return ordered ? <OrderedList>{children}</OrderedList> : <UnorderedList simple={simple}>{children}</UnorderedList>;
  }
}

export default TextList;
