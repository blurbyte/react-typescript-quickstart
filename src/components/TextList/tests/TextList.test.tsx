import * as React from 'react';
import { shallow } from 'enzyme';

import TextList from '../index';

describe('TextList', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<TextList />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when ordered', () => {
    const wrapper = shallow(<TextList ordered />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly list Item', () => {
    const wrapper = shallow(<TextList.Item />);
    expect(wrapper).toMatchSnapshot();
  });
});
