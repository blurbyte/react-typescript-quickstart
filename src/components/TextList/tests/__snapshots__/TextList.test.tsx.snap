// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextList renders correctly 1`] = `<styled.ul />`;

exports[`TextList renders correctly list Item 1`] = `
.c0 {
  margin: 0.2rem 0;
}

<styled.li
  className="c0"
/>
`;

exports[`TextList renders correctly when ordered 1`] = `<Styled(styled.ol) />`;
